automaton.py

```diff
@@ -1,37 +1,140 @@
+import enum
+
 class Transition():
 
-    def __init__(self, dest: str, target: str, symbol: str):
+    def __init__(self, dest: str, target: list=[], symbol: list=[]):
         self.dest = dest
         self.target = target
         self.symbol = symbol
 
     def render(self):
         # TODO: Return a string in the format s0:a>s1
         pass
+        
+class operationID(enum.Enum):
+    INVALID = 0
+    STATES = 1
+    INITIAL = 2
+    ACCEPT = 3
+    ALPHABET = 4
+    TRANSITIONS = 5
+    
+    
 
 class Automaton():
 
     def __init__(self, states: list = [], initial: str = "", accept: list = [], alphabet: list = [], transitions: list=[], from_file: str = ""):
-        self.states = states
-        self.initial = initial
-        self.accept = accept
-        self.alphabet = alphabet
-        self.transitions = transitions
+        self.states = []
+        self.initial = ""
+        self.accept = []
+        #include "$" (epsilon transition) in alphabet
+        self.alphabet = ['$']
+        self.transitions = []
         self.is_valid = False
         self.errors = []
         if from_file:
+            self.initial = ""
             self._parse_file(from_file)
         self.validate()
 
     def validate(self):
         # TODO: Validate if automaton is valid, if so, set self.is_valid = True
         # if not, add errors to self.errors
-        pass
+        
+        #"set" help to see if an element is repeated (if lenght is different from the lenght of the set, this means a element is repeated)
+        #states repeated, throw an error
+        if len(self.states) != len(set(self.states)):
```

Comment:
We can avoid checking repetition if we change the data type of states, accept and alphabet to `set`. WE can do the same for transitions, but we also need to implement the `==` operator in the class since class instances are different in memory space.

Reply:
Not necessarily to be implemented at this point, it can wait for a later commit